onellm/files.py

```
@@ -49,24 +55,81 @@ def upload(
+                name="file data"
+            )
+            file_data = file
+            filename = kwargs.pop("filename", "file.bin")
+            
+        elif hasattr(file, "read"):
+            # File-like object - read and validate
+            file_data = file.read()
+            if not isinstance(file_data, bytes):
+                raise InvalidRequestError(
+                    f"File object must read bytes, got {type(file_data).__name__}"
+                )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)